pkg/policy/policies_iterator.go

```diff
@@ -0,0 +1,54 @@
+package policy
+
+import (
+	"github.com/aquasecurity/tracee/pkg/utils"
+)
+
+// PoliciesIterator is an iterator for Policies.
+type PoliciesIterator struct {
+	policies []*Policy
+	index    int
+}
+
+// HasNext returns true if there are more policies to iterate.
+func (i *PoliciesIterator) HasNext() bool {
+	return i.index < len(i.policies)
+}
+
+// GetNext returns the next policy in the iteration.
+func (i *PoliciesIterator) GetNext() *Policy {
+	if i.HasNext() {
+		p := i.policies[i.index]
+		i.index++
+		return p
+	}
+
+	return nil
+}
+
+// CreateUserlandIterator returns a new iterator for a reduced list of policies
+// which must be filtered in userland (ArgFilter, RetFilter, ContextFilter,
+// UIDFilter and PIDFilter).
+func (ps *Policies) CreateUserlandIterator() utils.Iterator[*Policy] {
+	return &PoliciesIterator{
+		policies: ps.userlandPolicies,
+		index:    0,
+	}
+}
+
+// CreateAllIterator returns a new iterator for all policies.
+//
+// This iterator should not be used in hot paths since to avoid keeping
```

Comment:
WDYT?
```suggestion
// WARNING: This iterator should not be used in hot paths. In order to avoid keeping
// the list of policies in memory, create the iterator on the fly from its
// respective map instead.
```

Reply:
BTW, how bad is the memory footprint that we shouldn't keep the list in memory?

Reply:
256 * policy = up to 16MB... for each policies version in memory.

If better performance is required for that use, in the future we can keep that list static.

Reply:
In hindsight, I think I'll make it static right now.